### PR TITLE
Refactor api into separate module

### LINKED ISSUE #9: FIx problems with stale cli option values in tests
When testing in Go, all the tests are running in the same application context, so when setting a global command line flag, it will impact subsequent tests. This causes problems for things like #8, because e.g. a stale "quota" setting will change the generated output in subsequent tests. Rather than munging the expected output to account for this, we should figure out how to properly isolate the tests.

### PR DESCRIPTION
**Pull request by larsks**
Make unit testing easier by separating the API from the command-line
processing logic. There is now an `api` module that defines an API type,
and API methods are called on objects of that type.

Cobra commands are allocated dynamically, rather than statically. If
we re-introduce tests specifically for the CLI, this will make testing
easier by ensuring that each test can start with a clean Cobra
environment.

Closes #9

### CODE REVIEW

**Review by rob-baron — COMMENTED** [review]
Seems ok to me.  Looks like an incremental improvement. I approve.

**Review by rob-baron — APPROVED** [review]

**Review by tumido — COMMENTED** [review]
/lgtm

File: main.go [hunk 1]
```diff
@@ -8,8 +8,13 @@ package main
...
 	"github.com/operate-first/opfcli/cmd"
+	"github.com/operate-first/opfcli/utils"
+	"github.com/spf13/cobra"
 )
 
 func main() {
-	cmd.Execute()
+	utils.ConfigureLogging()
+
+	cmd := cmd.NewCmdRoot()
```
Comment (HumairAK) [inline main.go]: can you rename `cmd` here to something else so it's not colliding with the imported pkg?

File: cmd/main.go [hunk 2]
```diff
@@ -2,115 +2,108 @@
...
-			if err != nil {
-				repoDirectory, err = filepath.Abs(".")
-				if err != nil {
-					log.Fatalf("failed to determine repository directory: %v", err)
-				}
-			}
-		}
-		log.Debugf("using %s as repository directory", repoDirectory)
+func NewCmdRoot() *cobra.Command {
+	api := &api.API{}
```
Comment (HumairAK) [inline cmd/main.go]: can you rename `api` here so it doesn't collide with the imported pkg?

File: cmd/version.go [hunk 3]
```diff
@@ -3,24 +3,25 @@ package cmd
...
-	SilenceErrors: true,
-	RunE: func(cmd *cobra.Command, args []string) error {
-		fmt.Printf("Name: %s\n", version.Name)
-		fmt.Printf("Version: %s\n", version.BuildVersion)
-		fmt.Printf("Git Commit Hash: %s\n", version.BuildHash)
-		fmt.Printf("Build Date: %s\n", version.BuildDate)
-		return nil
-	},
-}
+func NewCmdVersion(api *api.API) *cobra.Command {
```
Comment (HumairAK) [inline cmd/version.go]: this function doesn't seem like it uses `api`

**Comment by HumairAK** [comment]
few nitpicks above, but other then that lgtm, I see that the quota flags are added in this pr w/o the logic, but that seems to come with the next pr so I think it's fine to leave it in.

**Review by 4n4nd — COMMENTED** [review]
/lgtm

File: cmd/version.go (hunk 3 above)
Comment (larsks) [reply cmd/version.go]: It's doesn't (right now), but this is effectively the standard function signature for new subcommands. I'm inclined to leave this as-is.

**Comment by larsks** [comment]
> I see that the quota flags are added in this pr w/o the logic, but that seems to come with the next pr so I think it's fine to leave it in.

Whoops, rebasing error. I agree re: leaving it in. Especially because it's only the README, and not actually in the code.

File: cmd/main.go (hunk 2 above)
Comment (larsks) [reply cmd/main.go]: ...and everywhere else, too, I guess.

**Review by tumido — COMMENTED** [review]
/lgtm

**Review by HumairAK — APPROVED** [review]
/lgtm

**Pull request merged**